### ISSUE TITLE
Create GraphQLHttpWebSocket only when needed

### ISSUE THREAD

**Issue by christianheld** [graphql-dotnet/graphql-client#246]
`GraphQLHttpClient` instance always creates a `GraphQLHttpWebSocket` instance, even if `Options.UseWebSocketForQueriesAndMutations` is false.

I guess the `GraphQLHttpWebSocket`'s subscription prevents GC from cleaning up client instances that have not been disposes manually (which should be done anyway).

Consider creating `GraphQLHttpWebSocket` on first use or only if `UseWebSocketForQueriesAndMutations` is true.

**Comment by Maydayof** [comment]
Based on the idea of christianheld I have modified locally the creation of the GraphQLHttpWebSocket (and I had to add the System.Reactive package), but with this modification I can use this graphql client in a Blazor WASM application, the server is HotChocolate asp.net core. (I have only tested queries, but I hope, the mutations will work fine also).
I can create GraphQLHttpClient instances on razor pages, but I can inject it into the pages also.

So, as christinaheld asked, the conditional creation would be nice, with it, and with some minor modification this client can run in Blazor WASM environment.

dotnet 3.1.300
VS 2019 16.6.0
GraphQL.Client 3.1.2
